native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/QueryProcessor.java

```diff
@@ -228,6 +219,43 @@ private static Object nativeQueryRowExecutable(
         return ErrorGenerator.getSQLApplicationError("Client is not properly initialized!");
     }
 
+    private static Object getUnionType(
+            UnionType describingType, ResultSet resultSet, AbstractResultParameterProcessor resultParameterProcessor)
+            throws SQLException, TypeMismatchError {
+        for (Type type: describingType.getMemberTypes()) {
+            try {
+                // If one of the types inside the union is a union, recursively check
+                if (type.getTag() == TypeTags.UNION_TAG) {
+                    return getUnionType(describingType, resultSet, resultParameterProcessor);
+                }
+                return getRecordOrPrimitiveType(type, resultSet, resultParameterProcessor);
+            } catch (ApplicationError e) {
+                // Ignored
```

Comment:
Why is the error ignored?

Reply:
This methods iterates through each type in the union type and checks whether it is compatible with the result from the query.

I checked the compatibility by trying the convert the result into each type (can be record or primitive), if an application error is thrown, that means it's incompatible, so it moves on to the next type. The error is ignored because it is expected. 

If all types are exhausted, a TypeMismatchError is thrown.

Reply:
Ack. Let's improve the comment to include these details.